stories/__tests__/SQFormGuidedWorkflow.stories.test.js

```diff
@@ -0,0 +1,249 @@
+import React from 'react';
+import {composeStories} from '@storybook/testing-react';
+import {render, screen, waitFor} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as stories from '../SQFormGuidedWorkflow.stories';
+
+const {Default: SQFormGuidedWorkflow, Testing} = composeStories(stories);
+
+describe('SQFormGuidedWorkflow Tests', () => {
+  it('should render a title, subtitle, and 3 sections', () => {
+    render(<SQFormGuidedWorkflow />);
+
+    const mainTitle = screen.getByText('CCA Guided Workflow');
+    expect(mainTitle).toBeVisible();
+
+    const mainSubtitle = screen.getByText(/please review/i);
+    expect(mainSubtitle).toBeVisible();
+
+    const openButton = screen.getByRole('button', {name: /open/i});
+    expect(openButton).toBeInTheDocument();
+
+    const toggleButtons = screen.getAllByRole('button', {
+      name: /toggle-expansion/i
+    });
+    expect(toggleButtons.length).toBe(3); // x 3
+
+    //First section defaults to open
+    const introSection = screen.getByRole('button', {
+      name: /toggle-expansion/i,
+      expanded: true
+    });
+    expect(introSection).toBeVisible();
+
+    //Other 2 sections not open
+    const remainingSections = screen.getAllByRole('button', {
+      name: /toggle-expansion/i,
+      expanded: false
+    });
+    expect(remainingSections.length).toBe(2);
+
+    const agentScript = screen.getByRole('heading', {name: /agent script/i});
+    expect(agentScript).toBeVisible();
+
+    //Should we also check for the icon and script text?
+
+    const outcomeArea = screen.getByRole('heading', {name: /confirm info/i});
+    expect(outcomeArea).toBeVisible();
+
+    //Should we also check for the dropdown and textarea?
+
+    const resetButton = screen.getByRole('button', {name: /reset form/i});
+    expect(resetButton).toBeVisible();
+    expect(resetButton).toBeDisabled();
+
+    const nextButton = screen.getByRole('button', {name: /form submission/i});
+    expect(nextButton).toBeVisible();
+    expect(nextButton).toBeDisabled();
+  });
+
+  it('should enable next button when all required fields are filled out in the 1st section', () => {
+    render(<SQFormGuidedWorkflow />);
+
+    const nextButton = screen.getByRole('button', {name: /form submission/i});
+    expect(nextButton).toBeDisabled();
+
+    const outcome = screen.getByRole('button', {name: /outcome/i});
+    userEvent.click(outcome);
+
+    const interested = screen.getByText('Interested');
+    userEvent.click(interested);
+
+    expect(nextButton).toBeEnabled();
+  });
+
+  it('should close curent expanded card and open the next one when next button is clicked', async () => {
+    render(<SQFormGuidedWorkflow />);
+
+    const introText = screen.queryByText(/hi, bob smith, my name is/i);
```

Comment:
I believe this one should be `getByText` if we're testing that it's visible/exists

Reply:
I used `queryByText` so that I could reuse it again on line 101 to test that it's NOT there. If I use `getByText` and it's not there then it will error out and fail the test.